src/biconnectivity13/biconnect.jl

```diff
@@ -84,16 +27,68 @@ julia> biconnected_components(cycle_graph(5))
 """
 function biconnected_components end
 @traitfn function biconnected_components(g::::(!IsDirected))
-    state = Biconnections(g)
-    for u in vertices(g)
-        if state.depth[u] == 0
-            visit!(g, state, u, u)
-        end
+        n = nv(g)
+        S = Array{Tuple{Int, Int}, 1}(undef,n)
+        E = Edge{eltype(g)}
+        edge_st = Vector{E}()
+        components = Vector{Vector{E}}()
+        us = one(Int)
```

Comment:
This should not be `Int` if you're using it as an iterator to `vertices(g)` - why define it at all?